lib/ansible/modules/network/netvisor/pn_role.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/python
+# Copyright: (c) 2018, Pluribus Networks
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = """
+---
+module: pn_role
+author: "Pluribus Networks (@rajaspachipulusu17)"
+version_added: "2.8"
+short_description: CLI command to create/delete/modify role
+description:
+  - This module can be used to create, delete and modify user roles.
+options:
+  pn_cliswitch:
+    description:
+      - Target switch to run the CLI on.
+    required: false
+    type: str
+  state:
+    description:
+      - State the action to perform. Use C(present) to create role and
+        C(absent) to delete role and C(update) to modify role.
+    required: true
+    type: str
+    choices: ['present', 'absent', 'update']
+  pn_scope:
+    description:
+      - local or fabric.
+    required: false
+    type: str
+    choices: ['local', 'fabric']
+  pn_access:
+    description:
+      - type of access.
+    required: false
+    type: str
+    choices: ['read-only', 'read-write']
+  pn_shell:
+    description:
+      - allow shell command.
+    required: false
+    type: bool
+  pn_sudo:
+    description:
+      - allow sudo from shell.
+    required: false
+    type: bool
+  pn_running_config:
+    description:
+      - display running configuration of switch.
+    required: false
+    type: bool
+  pn_name:
+    description:
+      - role name.
+    required: true
+    type: str
+  pn_delete_from_users:
+    description:
+      - delete from users.
+    required: false
+    type: bool
+"""
+
+EXAMPLES = """
+- name: role functionality
+  pn_role:
+    pn_cliswitch: 'sw01'
+    state: 'present'
+    pn_name: 'foo'
+    pn_scope: 'local'
+    pn_access: 'read-only'
+
+- name: role functionality
```

Comment:
@rajaspachipulusu17 can u please update the `name` accordingly to what the play actually does as the current one have generic name.

Reply:
fixed it.